third-party/rsutils/include/rsutils/concurrency/concurrency.h

```diff
@@ -50,9 +50,10 @@ class single_consumer_queue
 
         if( _queue.size() > _cap )
         {
+            auto& q = _queue;
```

Comment:
I did not understand this change. What problem does it solve?

Reply:
this fix removed.
cppcheck complaints to variable scope as we pass pointer to cb function.

Reply:
any ideas?

Reply:
`concurrency.h` This is the heart of the SDK, please verify you both understand the change and that it is needed

Reply:
Seems like the problem is not in concurency but in the calling function [notification_model::invoke](https://github.com/IntelRealSense/librealsense/blob/master/common/notifications.cpp#L760).
Trying to check if an action was actually invoked (within timeout of 100 seconds) they use reference to a local object `q` that might be used after the function has returned.
i think that we should clear `_dispatch_queue` in case of timeout to make sure reference to `q` won't be used later...

Reply:
@OhadMeir lets speak about it

Reply:
What about this cppcheck issue? Going to fix [notificaiton_model::invoke](https://github.com/IntelRealSense/librealsense/blob/master/common/notifications.cpp#L760) or on a different PR?

Reply:
Fixed